### DISCUSSION TITLE
send RouterData with navigate()

### DISCUSSION THREAD

**Discussion by ghost** [solidjs/solid-router#211] [Ideas]
let's say we are at `/step/2`, and we're about to go to `/step/3`
and we pass `userAge` as RouterData 
```ts
navigate("/step/3", ...., { userAge: 18 } )
```

basically passing RouterData to route with navigate..

good or bad? If good then could you implement
if bad tell why, do you have alternative which is not that complicated

**Comment by outbackStack** [comment]
what do you mean good or bad? 

You can use this to pass the state to `/step/3` and access it with
`const location = useLocation();` in the new route
